modelscan/modelscan.py

```diff
@@ -71,127 +71,131 @@ def scan(
         self._skipped = []
         self._scanned = []
         self._input_path = str(path)
-        pathlibPath = Path().cwd() if path == "." else Path(path).absolute()
-        self._scan_path(Path(pathlibPath))
+        pathlib_path = Path().cwd() if path == "." else Path(path).absolute()
+        model_path = Path(pathlib_path)
+
+        try:
+            with Model.from_path(model_path) as model:
+                self._scan_model(model)
+        except ModelPathNotValid as e:
+            logger.exception(e)
+            self._errors.append(
+                ModelScanError(
+                    "ModelScan", ErrorCategories.PATH, "Path is not valid", str(path)
+                )
+            )
+        except ModelIsDir:
```

Comment:
There is no scanner using folder scanning, so we can do it as a fallback if Path upstream is not file